src/business-logic/services/timeout-callback-scheduler.ts

```diff
@@ -39,6 +39,7 @@ export class TimeoutCallbackScheduler implements CallbackScheduler {
     }
     const durationToNextSchedulingInMs: number = durationToExecuteCallbackInMs / SCHEDULE_RESOLUTION
     this.timeoutIdentifier = setTimeout(() => {
+      this.stop()
```

Comment:
This should be outside of the setTimeout callback, before the timeoutIdentifier value is overwritten, since this can lead to incorrect behaviour. As I understand it, if a new callback is scheduled, the old timeout should be cleared and the one should be kept.
Since `this.stop()` clears the timer tied to `this.timeoutIdentifier`, then it will always be the latest scheduled callback that gets cleared. It is not an issue in the RundownTimelineService, since we manually trigger the stop method when calling takeNext.

Since this can be a bit tricky to see through, it would be nice with a few tests for this case :)
E.g. if the first logarithmic step from the second scheduled callback triggers before the first scheduled callback, then both callbacks will reschedule and none of the callbacks will be stopped.

Reply:
Updated